var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -230,7 +230,18 @@ def install(self, spec, prefix):
         # appear twice in in link line but this is harmless
         lapack_blas = spec['lapack'].libs + spec['blas'].libs
 
-        options.append('BLAS_LIBS={0}'.format(lapack_blas.ld_flags))
+        # qe-6.5 fails to detect MKL for FFT if BLAS_LIBS is set due to
+        # an unfortunate upsteam change in their autoconf/configure:
+        # - qe-6.5/install/m4/x_ac_qe_blas.m4 only sets 'have_blas'
+        #   but no 'have_mkl' if BLAS_LIBS is set (which seems to be o.k.)
+        # - however, qe-6.5/install/m4/x_ac_qe_fft.m4 in 6.5 unfortunately
+        #   relies on x_ac_qe_blas.m4 to detect MKL and set 'have_mkl'
+        # - qe-5.4 up to 6.4.1 had a different logic and worked fine with
+        #   BLAS_LIBS being set
+        # However, MKL is correctly picked up by qe-6.5 for BLAS and FFT if
+        # MKLROOT is set (which SPACK does automatically for ^intel-mkl)
+        if not ('quantum-espresso@6.5' in spec and '^intel-mkl' in spec):
```

Comment:
```suggestion
        if not ('quantum-espresso@6.5' in spec and '^mkl' in spec):
```
This will work for both the `intel-mkl` and `intel-parallel-studio+mkl` packages that provide MKL libraries.

Reply:
You are right!
I not only changed '^intel-mkl' to '^mkl' here, but throughout the file with commit 5811e34.